syntax = "proto3";

package fractal;

message Point {
  double real = 1;
  double img = 2;
}

message Header {
  int32 signature = 1; // [default = 0x46520100 ];
  Point bb_tl = 3;
  Point bb_br = 4;
  int32 real_samples = 5;
  int32 img_samples = 6;
  int32 limit = 7;
  int32 max_iterations = 8;
  int32 min_iterations = 9;
}

message Results {
  Point  last_value = 1;
  double last_modulus = 2;
  int32  iterations = 3;
  bool   diverged = 4;
}

// should have real_samples # of results in a row
// should have img_samples # of rows in the file
message Row {
  repeated Results results = 2;
}
